src/test/kotlin/com/intellij/ml/llm/template/BatchProcessingExtractFunction.kt

```diff
@@ -0,0 +1,186 @@
+package com.intellij.ml.llm.template
+import com.intellij.ml.llm.template.models.GPTExtractFunctionRequestProvider
+import com.intellij.ml.llm.template.models.LLMRequestProvider
+import com.intellij.ml.llm.template.utils.*
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiNameIdentifierOwner
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.testFramework.LightPlatformCodeInsightTestCase
+import com.mongodb.client.MongoClient
+import com.mongodb.client.MongoClients
+import org.bson.*
+import org.junit.jupiter.api.TestInstance
+
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class BatchProcessingExtractFunction : LightPlatformCodeInsightTestCase() {
+    private val efLLMRequestProvider: LLMRequestProvider = GPTExtractFunctionRequestProvider
+    private var mongoClient: MongoClient = MongoClients.create("mongodb://localhost:27017")
+    private var projectPath = "src/test"
+
+    override fun getTestDataPath(): String {
+        return projectPath
+    }
+
+    fun testNothing() {
+        println("testing nothing")
+    }
+
+//    fun `test gather extract function suggestions`() {
```

Comment:
In this case we do. I commented it because it's not supposed to run as a unit test. This eventually will go in a separate library, but for now I made it as a UT so that it can have access to the code that does the Extract Function transformation

Reply:
Maybe we can create a file in `resources` folder and move it there?